Build with the poetry-core backend, the legacy poetry one is gone.

diff -Nrup a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2022-08-26 17:30:59.000000000 +0200
+++ b/pyproject.toml	2026-10-22 17:22:16.291974812 +0200
@@ -15,7 +15,7 @@ update-copyright = ">=0.6.2"
 
 [build-system]
 requires = ["poetry>=0.12"]
-build-backend = "poetry.masonry.api"
+build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.scripts]
 r2e = "rss2email.main:run"
